src/freenet/support/io/Fallocate.java

```diff
@@ -111,4 +118,28 @@ private static int getDescriptor(FileDescriptor descriptor) {
       throw new UnsupportedOperationException("unsupported FileDescriptor implementation", e);
     }
   }
+
+  public static void legacyFill(FileChannel fc, long newLength, long offset, boolean starting) throws IOException {
+    MersenneTwister mt = new MersenneTwister();
+    byte[] b = new byte[4096];
+    ByteBuffer bb = ByteBuffer.wrap(b);
+    int x = 0;
+    while (offset < newLength) {
+      bb.rewind();
+      mt.nextBytes(b);
+      offset += fc.write(bb, offset);
+      if (offset % (1024 * 1024 * 1024L) == 0) {
+        mt = new MersenneTwister();
+        if (starting) {
+          WrapperManager.signalStarting( (int) MINUTES.toMillis(5));
+          if (x++ % 32 == 0)
+            System.err.println(
+                "Preallocating space : "
+                + offset
+                + "/"
+                + newLength);
```

Comment:
Alright, that sort of makes sense. Once per 32 GiB as it is now is not too
bad, changing that to % 32 = 1 or similar would help to avoid displaying it
for small resizes.
Op 8 jul. 2016 00:58 schreef "Florent Daigniere" <notifications@github.com>:

In src/freenet/support/io/Fallocate.java
<https://github.com/freenet/fred/pull/575#discussion_r70002452>:

> +    ByteBuffer bb = ByteBuffer.wrap(b);
> +    int x = 0;
> +    while (offset < newLength) {
> +      bb.rewind();
> +      mt.nextBytes(b);
> +      offset += fc.write(bb, offset);
> +      if (offset % (1024 * 1024 * 1024L) == 0) {
> +        mt = new MersenneTwister();
> +        if (starting) {
> +          WrapperManager.signalStarting( (int) MINUTES.toMillis(5));
> +          if (x++ % 32 == 0)
> +            System.err.println(
> +                "Preallocating space : "
> +                + offset
> +                + "/"
> +                + newLength);

toad wanted it to be visible in wrapper.log since that's the only thing
users can be expected to see; I can reduce the periodicity if you want...
what would be reasonable?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/freenet/fred/pull/575/files/548f081d4ed3f62f2d1b83eaebcc9dffc1d79f23..3f5291a7bd2ab96d77417dd12fbd08f50ea5529c#r70002452>,
or mute the thread
<https://github.com/notifications/unsubscribe/AATu91YLWJ_5SGOfriajGXzn6F9aj-E3ks5qTYSqgaJpZM4JFRFm>
.